nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -353,7 +386,44 @@ def validation_step(self, batch, batch_idx):
         if loss_mean.item == 0.0:
             loss_mean = []
 
-        return loss_mean
+        if self.cfg.get('report_validation_metric', False):
+            preds_text, labels_text = [], []
```

Comment:
Maybe declare these variables much lower (ie close to line 413) See https://softwareengineering.stackexchange.com/questions/113262/why-declare-variables-close-to-where-they-are-used